tests/periph_timer_periodic/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Beuth Hochschule für Technik Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Periodic timer test application
+ *
+ * @author      Benjamin Valentin <benpicco@beuth-hochschule.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include "board.h"
+
+#include "mutex.h"
+#include "periph/timer.h"
+
+/* the timer device to use */
+#define TIMER_CYCL  (0)
+
+static unsigned count[TIMER_CHANNELS];
+
+static void cb(void *arg, int chan)
+{
+    int c = count[chan]++;
+
+    if ((c & 0x3) == 0) {
+        printf("[%d] tick\n", chan);
+    }
+
+    if (c > 20) {
+        timer_stop(TIMER_CYCL);
+        mutex_unlock(arg);
+    }
+}
+
+static const char* _print_ok(int chan, bool *succeeded)
+{
+    if (chan == 0 && count[chan] > 0) {
+        return "OK";
+    }
+
+    if (chan > 0 && count[chan] == 0) {
+        return "OK";
+    }
+
+    *succeeded = false;
+    return "ERROR";
+}
+
+int main(void)
+{
+    mutex_t lock = MUTEX_INIT_LOCKED;
+    const unsigned timer_hz = 62500;
+    const unsigned steps = (250UL * timer_hz) / 1000; /* 250 ms */
+
+    printf("\nRunning Timer %d at %u Hz.\n", TIMER_CYCL, timer_hz);
+    printf("One counter cycle is %u ticks or 250 ms\n", steps);
+    puts("Will print 'tick' every second / every 4 cycles.\n");
+
+    assert(timer_init(TIMER_CYCL, timer_hz, cb, &lock) == 0);
```

Comment:
Note that with `NDEBUG`, the timer ends up being uninitialized. That what `expect()` is for